test_times.py

```diff
@@ -0,0 +1,18 @@
+from times import *
+
+def test_given_input():
+    large = time_range("2010-01-12 10:00:00", "2010-01-12 12:00:00")
+    short = time_range("2010-01-12 10:30:00", "2010-01-12 10:45:00", 2, 60)
+
+    result = large
+    expected = [('2010-01-12 10:00:00', '2010-01-12 12:00:00')]
+    assert result == expected
```

Comment:
Nice - Makes a lot of sense to test `time_range` too!
I'd suggest to separate the test(s) for `time_range` from the test(s) for `compute_overlap` into separate unit tests, because that might make it easier to know where to look if one of the tests fails?